pkg/windows/windows.go

```diff
@@ -166,11 +165,24 @@ var (
 	}
 )
 
-// getFilesToTransfer returns the properly populated filesToTransfer map. Note this does not include the WICD binary.
-func getFilesToTransfer() (map[*payload.FileInfo]string, error) {
-	if filesToTransfer != nil {
-		return filesToTransfer, nil
+func createFileInfo(platform *config.PlatformType) (map[*payload.FileInfo]string, error) {
+	srcDestPairs, err := getFilesToTransfer(platform)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get list of files to transfer: %w", err)
 	}
+	files := make(map[*payload.FileInfo]string)
+	for src, dest := range srcDestPairs {
+		f, err := payload.NewFileInfo(src)
+		if err != nil {
+			return nil, fmt.Errorf("could not create FileInfo object for file %s: %w", src, err)
+		}
+		files[f] = dest
+	}
+	return files, nil
+}
+
+// getFilesToTransfer returns the properly populated filesToTransfer map. Note this does not include the WICD binary.
+func getFilesToTransfer(platform *config.PlatformType) (map[string]string, error) {
```

Comment:
The restructuring you are doing is hinting that this method also needs to move to the `payload` package. The way to do it is have a pre-commit that moves it and then make changes in a follow up commit.

Reply:
if we move the getFilesToTransfer function to payload, we'd have to move all the constants that are in the windows package 
is that a good idea?

Reply:
The refactor also needs to take of the cyclic dependency that will get introduced. Given the turnaround time it is taking, lets leave it here for now.